rules/no-array-reduce.js

```diff
@@ -104,8 +106,7 @@ const create = context => {
 				const methodNode = getMethodNode(callExpression);
 				yield {
 					node: methodNode,
-					messageId: MESSAGE_ID,
-					data: {method: methodNode.name},
+					messageId: methodNode.name,
```

Comment:
Pattern seen in https://github.com/sindresorhus/eslint-plugin-unicorn/blob/1558cbea81bf4476ee27d80724155d991433ce49/rules/prefer-string-slice.js#L141